library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java

```diff
@@ -1078,6 +1078,33 @@ public void testSelectTracksWithinCapabilitiesAndForceLowestBitrateSelectLowerBi
     assertThat(result.selections.get(0).getSelectedFormat()).isEqualTo(lowerBitrateFormat);
   }
 
+  /**
+   * Tests that track selector will select audio tracks with higher bitrate when {@link Parameters}
+   * indicate highest bitrate preference, even when tracks are within capabilities.
+   */
+  @Test
+  public void testSelectTracksWithinCapabilitiesAndForceHighestBitrateSelectHigherBitrate()
+          throws Exception {
+    trackSelector.setParameters(new ParametersBuilder().setForceHighestSupportedBitrate(true).build());
+
+    Format lowerBitrateFormat =
+            Format.createAudioSampleFormat("audioFormat", MimeTypes.AUDIO_AAC, null, 15000,
+                    Format.NO_VALUE, 2, 44100, null, null, 0, null);
+    Format higherBitrateFormat =
+            Format.createAudioSampleFormat("audioFormat", MimeTypes.AUDIO_AAC, null, 30000,
+                    Format.NO_VALUE, 2, 44100, null, null, 0, null);
+
+    TrackSelectorResult result =
+            trackSelector.selectTracks(
+                    new RendererCapabilities[] {ALL_AUDIO_FORMAT_SUPPORTED_RENDERER_CAPABILITIES},
+                    singleTrackGroup(lowerBitrateFormat, higherBitrateFormat),
+                    periodId,
+                    TIMELINE);
+
+    assertThat(result.selections.get(0).getSelectedFormat()).isEqualTo(higherBitrateFormat);
```

Comment:
You should probably add an assertion that only one track is in the selection:
assertThat(result.selections.get(0).length()).isEqualTo(1);

Or to put it the other way round: The test should fail if you comment out the first line (which I think is currently not the case).

Reply:
The tracks will be sorted by decending bitrate, so having the lowest bitrate as the first track means there can be only one track. That's very non-obvious though and I agree we should add that to testSelectTracksWithinCapabilitiesAndForceLowestBitrateSelectLowerBitrate as well.